statistics/handle/update.go

```diff
@@ -998,12 +998,12 @@ var AutoAnalyzeMinCnt int64 = 1000
 // TableAnalyzed checks if the table is analyzed.
 func TableAnalyzed(tbl *statistics.Table) bool {
 	for _, col := range tbl.Columns {
-		if col.Count > 0 {
```

Comment:
The condition is not precise. Since a newly-added column has generated stats according to its default value but it may be not analyzed.